api/datadoghq/v2alpha1/datadogagent_types.go

```diff
@@ -170,6 +170,88 @@ type SingleStepInstrumentation struct {
 	// Injector configures the APM Injector.
 	// +optional
 	Injector *InjectorConfig `json:"injector,omitempty"`
+
+	// Targets is a list of targets to apply the auto instrumentation to. The first target that matches the pod will be
+	// used. If no target matches, the auto instrumentation will not be applied. Full config key:
+	// apm_config.instrumentation.targets
+	// (Requires Cluster Agent 7.64.0+)
+	// +optional
+	Targets []SSITarget `json:"targets,omitempty"`
+}
+
+// SSITarget is a rule to apply the auto instrumentation to a specific workload using the pod and namespace selectors.
+// Full config key: apm_config.instrumentation.targets to get the list of targets.
+type SSITarget struct {
+	// Name is the name of the target. It will be appended to the pod annotations to identify the target that was used.
+	// Full config key: apm_config.instrumentation.targets[].name
+	Name string `json:"name"`
+	// PodSelector is the pod selector to match the pods to apply the auto instrumentation to. It will be used in
+	// conjunction with the NamespaceSelector to match the pods. Full config key:
+	// apm_config.instrumentation.targets[].selector
+	PodSelector PodSelector `json:"podSelector"`
+	// NamespaceSelector is the namespace selector to match the namespaces to apply the auto instrumentation to. It will
+	// be used in conjunction with the Selector to match the pods. Full config key:
+	// apm_config.instrumentation.targets[].namespaceSelector
+	NamespaceSelector NamespaceSelector `json:"namespaceSelector"`
+	// TracerVersions is a map of tracer versions to inject for workloads that match the target. The key is the tracer
+	// name and the value is the version to inject. Full config key:
+	// apm_config.instrumentation.targets[].ddTraceVersions
+	TracerVersions map[string]string `json:"ddTraceVersions"`
+	// TracerConfigs is a list of configuration options to use for the installed tracers. These options will be added
+	// as environment variables in addition to the injected tracer. Full config key:
+	// apm_config.instrumentation.targets[].ddTraceConfigs
+	TracerConfigs []TracerConfig `json:"ddTraceConfigs"`
+}
+
+// PodSelector is a reconstruction of the metav1.LabelSelector struct to be able to unmarshal the configuration. It
+// can be converted to a metav1.LabelSelector using the AsLabelSelector method. Full config key:
+// apm_config.instrumentation.targets[].selector
+type PodSelector struct {
```

Comment:
could we reuse this type? https://github.com/DataDog/datadog-operator/blob/main/api/datadoghq/v2alpha1/datadogagent_types.go#L869-L877

Reply:
We've used our own types top be able to control the struct tags attached. These also are used in the agent here, and we're trying to keep the types mirrored: https://github.com/DataDog/datadog-agent/blob/87919c66953aac6457eccc55af944684b41be074/pkg/clusteragent/admission/mutate/autoinstrumentation/config.go#L206